src/tz390.java

```diff
@@ -7684,4 +7688,36 @@ public boolean check_java_version(){
 		}
 		return true;
 }
+
+public String getVersion() {
+
+    String result = "Uncontrolled";
+
+    try (InputStream input = getClass().getClassLoader().getResourceAsStream("z390.properties")) {
+
+        Properties prop = new Properties();
+
+        if (input != null) {
+            prop.load(input);
+            result = prop.getProperty("version");
+            if (result == null) {
+                System.out.println("Unable to set version - property version not set");
+                result = "Uncontrolled";                
+            } else if (result.isEmpty()) {
+                System.out.println("Unable to set version - no value set");
+                result = "Uncontrolled";
+            } 
+        } else {
+            System.out.println("Unable to set version - z390.properties file not found");
+        }
+
+    } catch (IOException ex) {
+        System.out.println("Unable to set version - unexpected exception");
+        result =  "Uncontrolled";
+        ex.printStackTrace();
+    }
+    
+    return result;
+
+}
```

Comment:
It may be overkill, but it would help debugging if the setting of result to "Uncontrolled" on lines 7705, 7708, and 7716 were "Uncoltrolled-1", "Uncontrolled-2", and "Uncontrolled-3", respectively. Note that result is implicitly set to "Uncontrolled" at the "else" on line 7710, so the value of version would indicate exactly which error occurred. (The System.out.println output would probably be long gone by the time anyone noticed the error.)

Reply:
No probs to link the version to the error. 

I have been working on enhancing the relver script so that if does the checking at compile time rather than run time. The check in java is runtime.. by the time you get here, it is too late.

Here is what I am thinking.. 
When called from bldjar, it would stop build if git not found, but provide a message to say you can ignore by providing the parameter force to bldjar. This would require passing $1 to the relver script, but thats it.
The dev would need to provide force and be told why a version is not provded and can continue to build without git installed. Version is set by script rather than Java.. so the errors above are more related to an incorrect build process.
Let me know what you think as an alternative.. 

``` bash
set +e                  # need to handle errors in this script
cd $(dirname $0)        # set to directory of script
cd ..

VER="Uncontrolled";

# is git installed
$(git --version > /dev/null 2>&1)
GIT_INSTALLED=$?
if [ ${GIT_INSTALLED} != 0 ]; then
    echo >&2 "Git is not installed";
else
    # is in a git repo
    $(git rev-parse --is-inside-work-tree > /dev/null 2>&1)
    GIT_REPO=$?
    if [ ${GIT_REPO} != 0 ]; then
        echo >&2 "Not a git repo.";
    else
        VER=$(git describe);
    fi    
fi

set -e

if [[ ${VER} == "Uncontrolled" ]]; then
    parm=$(echo "$1" | tr '[:lower:]' '[:upper:]');
    if [[ "$parm" != "FORCE" ]]; then
        echo "Version is uncontrolled - add FORCE parm to ignore"
        exit 8;
    fi
fi
echo "version=${VER}" > z390.properties
cat z390.properties;

```

Reply:
The `bldjar` script now fails if run without git. You can still compile by running `bldjar force` but the build version will be set to "Uncontrolled". The reason for the error will be provided at this point. Both Win bat and Bash versions have been updated.
* Windows bc736e9
* Bash 2b4fb95

I have also added debugging in case someone does manage to compile and not use the bldjar script and one of the errors in TZ390 occur. If bldjar is used, then these errors should never happen.. see 3e45ca0